ingress/controllers/nginx/nginx/command.go

```diff
@@ -128,7 +128,10 @@ func (ngx Manager) testTemplate(cfg []byte) error {
 		return err
 	}
 	defer tmpfile.Close()
-	defer os.Remove(tmpfile.Name())
+	// do not remove temporal nginx.conf file if log level is >= 3
+	if !glog.V(3) {
```

Comment:
change the behavior to remove the file only if there's no error running the check